packages/aws-cdk-lib/core/test/removal-policies.test.ts

```diff
@@ -220,4 +220,18 @@ describe('removal-policys', () => {
     expect(bucket.cfnOptions.deletionPolicy).toBe('Retain');
     expect(table.cfnOptions.deletionPolicy).toBe('RetainExceptOnCreate');
   });
+
+  test('higher priority removal policy overrides lower priority removal policy', () => {
```

Comment:
It would be good to add a new test using `priority` and `overwrite` with true.

Reply:
Added.